### PR TITLE
Dump timeout

### LINKED ISSUE #5: Add support for dumping stack after a timeout
I currently have a test which hangs on Windows sometimes, and I have no idea why - so I thought about adding something to pytest-faulthandler which essentially does this:
- Add an argument `--faulthandler-timeout 60` or so - the traceback will be dumped after that many seconds.
- Before each test, call `faulthandler.dump_traceback_later(timeout)`
- After each test, call `faulthandler.cancel_dump_traceback_later()`

What do you think?

### PR DESCRIPTION
**Pull request by nicoddemus**
Fixes #5

### CODE REVIEW

**Comment by nicoddemus** [comment]
Would you mind taking a look and see if you feel there's anything missing @The-Compiler? :smile:

Thanks! :beers:

**Comment by The-Compiler** [comment]
Why imply `--capture=no`? You could simply pass the duplicated file handle to `dump_traceback_later`, just like with `enable`.

I'll take a look at the rest tomorrow :smile:

**Comment by nicoddemus** [comment]
> Why imply --capture=no? You could simply pass the duplicated file handle to dump_traceback_later, just like with enable.

Duh, of course. Brain fart, thanks for pointing it out. :stuck_out_tongue_closed_eyes: 

> I'll take a look at the rest tomorrow

No problem, no rush. Once you approve this I'll make a `1.0` release. :smile:

File: pytest_faulthandler.py [hunk 1]
```diff
@@ -8,12 +8,40 @@ def pytest_addoption(parser):
...
+                raise pytest.UsageError(msg)            
+                
 
+@pytest.mark.hookwrapper
+def pytest_runtest_protocol(item):
+    enabled = item.config.getoption('fault_handler')
+    timeout = item.config.getoption('fault_handler_timeout')
+    if enabled and timeout > 0:
+        import faulthandler
+        stderr = file=item.config.fault_handler_stderr
```
Comment (The-Compiler) [inline pytest_faulthandler.py]: I don't think the `file=` is intended here

File: test_pytest_faulthandler.py [hunk 2]
```diff
@@ -45,3 +48,46 @@ def test_disabled():
         "*1 passed*"
     ])
     assert result.ret == 0
+
+
+@pytest.mark.parametrize('enabled', [True, False])
+@pytest.mark.skipif(sys.platform.startswith('win'), reason='linux only')
+def test_timeout(testdir, enabled):
+    """Test option to dump tracebacks after a certain timeout (linux only).
+    If falthandler is disabled, no traceback will be dump.
```
Comment (The-Compiler) [inline test_pytest_faulthandler.py]: falthandler -> faulthandler; dump -> dumped :wink:

**Comment by The-Compiler** [comment]
LGTM apart from the comments - thank you!

**Pull request merged**